Add configuration option for `archive_source` (#337)
tealdeer-rs/tealdeer: docs/src/config_updates.md

    auto_update = true
    auto_update_interval_hours = 24

### archive_source

> URL for the location of tldr pages archive. By default the pages are

Reviewer: ```suggestion
URL for the location of the tldr pages archive. By default the pages are
```

> URL for the location of tldr pages archive. By default is the pages are
> fetched from the latest `tldr-pages/thdr` GitHub release.

Reviewer: ```suggestion
URL for the location of the tldr pages archive. By default the pages are
fetched from the latest `tldr-pages/tldr` GitHub release.
```

    auto_update = true
    auto_update_interval_hours = 24

> ### archive_url

Reviewer: Given that we will soon switch to using a URL where we append `pages.en.zip`, `pages.de.zip` etc. ourselves, maybe we should prefer something more neutral as the name. How do you feel about "archive_source"?

> URL for the location of tldr pages archive. Default is the main `thdr.sh`
> archive location.

Reviewer: The new client specification has changed the recommended source for pages again, instead of using `tldr.sh`, the Github releases are preferred now. I would thus change this to
```suggestion
URL for the location of tldr pages archive. By default the pages are fetched from the latest `tldr-pages/tldr` GitHub release.
```

    [updates]

>     auto_update = true
>     auto_update_interval_hours = 24

Reviewer: I think we should omit these here, because they are independent of the `archive_url` option
```suggestion
```

>     archive_url = https://tldr.infra.local/assets/tldr.zip

Reviewer: Let's go with something more neutral like
```suggestion
    archive_url = https://my-company.example.com/tldr.zip
```
